src/coq_elpi_HOAS.ml

```diff
@@ -1154,7 +1160,12 @@ let get_options ~depth hyps state =
       | Some true -> Some Heuristic
       | Some false -> Some Verbatim end;
     local = locality @@ get_string_option "coq:locality";
-    deprecation = deprecation @@ get_pair_option API.BuiltInData.string API.BuiltInData.string "coq:deprecation";
+    user_warns =
+      begin
+        let depr = deprecation @@ get_pair_option API.BuiltInData.string API.BuiltInData.string "coq:deprecation" in
+        let warn = warns @@ get_pair_option API.BuiltInData.string API.BuiltInData.string "coq:warns" in
+        Some UserWarn.{ depr; warn }
```

Comment:
This needs a macro and some doc. All options have one

Reply:
Thanks, done